src/time_integration/paired_explicit_runge_kutta/paired_explicit_runge_kutta.jl

```diff
@@ -5,8 +5,176 @@
+Add a time stop during the time integration process.
+This function is called after the periodic SaveSolutionCallback to specify the next stop to save the solution.
+"""
+function add_tstop!(integrator::AbstractPairedExplicitRKSingleIntegrator, t)
+    integrator.tdir * (t - integrator.t) < zero(integrator.t) &&
+        error("Tried to add a tstop that is behind the current time. This is strictly forbidden")
+    # We need to remove the first entry of tstops when a new entry is added.
+    # Otherwise, the simulation gets stuck at the previous tstop and dt is adjusted to zero.
+    if length(integrator.opts.tstops) > 1
+        pop!(integrator.opts.tstops)
+    end
+    push!(integrator.opts.tstops, integrator.tdir * t)
+end
+
+has_tstop(integrator::AbstractPairedExplicitRKSingleIntegrator) = !isempty(integrator.opts.tstops)
+first_tstop(integrator::AbstractPairedExplicitRKSingleIntegrator) = first(integrator.opts.tstops)
+
+# Fakes `solve`: https://diffeq.sciml.ai/v6.8/basics/overview/#Solving-the-Problems-1
+function solve(ode::ODEProblem, alg::AbstractPairedExplicitRKSingle;
+               dt, callback = nothing, kwargs...)
+    integrator = init(ode, alg, dt = dt, callback = callback; kwargs...)
+
+    # Start actual solve
+    solve!(integrator)
+end
+
+function solve!(integrator::PairedExplicitRK)
+    @unpack prob = integrator.sol
+
+    integrator.finalstep = false
+
+    @trixi_timeit timer() "main loop" while !integrator.finalstep
+        step!(integrator)
+    end # "main loop" timer
+
+    return TimeIntegratorSolution((first(prob.tspan), integrator.t),
+                                  (prob.u0, integrator.u),
+                                  integrator.sol.prob)
+end
+
+# used for AMR (Adaptive Mesh Refinement)
+function Base.resize!(integrator::PairedExplicitRK, new_size)
+    resize!(integrator.u, new_size)
+    resize!(integrator.du, new_size)
+    resize!(integrator.u_tmp, new_size)
+
+    resize!(integrator.k1, new_size)
+    resize!(integrator.k_higher, new_size)
+end
+
+# get a cache where the RHS can be stored
+get_du(integrator::PairedExplicitRK) = integrator.du
+get_tmp_cache(integrator::PairedExplicitRK) = (integrator.u_tmp,)
+
+# some algorithms from DiffEq like FSAL-ones need to be informed when a callback has modified u
+u_modified!(integrator::PairedExplicitRK, ::Bool) = false
+
+# used by adaptive timestepping algorithms in DiffEq
+function set_proposed_dt!(integrator::PairedExplicitRK, dt)
+    (integrator.dt = dt; integrator.dtcache = dt)
+end
+
+function get_proposed_dt(integrator::PairedExplicitRK)
+    return ifelse(integrator.opts.adaptive, integrator.dt, integrator.dtcache)
+end
+
+# stop the time integration
+function terminate!(integrator::PairedExplicitRK)
+    integrator.finalstep = true
+end
+
+"""
+    modify_dt_for_tstops!(integrator::PairedExplicitRK)
+Modify the time-step size to match the time stops specified in integrator.opts.tstops.
```

Comment:
```suggestion
    modify_dt_for_tstops!(integrator::PairedExplicitRK)
 
Modify the time-step size to match the time stops specified in integrator.opts.tstops.
```
I think a newline is needed to render this correctly.